File: backend/src/main/java/com/hst/sitescrapper/type/UserStatusType.java
package com.hst.sitescrapper.type;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

/**
 * @author dlgusrb0808@gmail.com
 */
public enum UserStatusType {
	EMAIL_NOT_VERIFIED(1, "이메일 미 인증 상태"),
	INACTIVE		  (2, "비활성 상태"),
	ACTIVE			  (3, "활성 상태");

	private int statusCode;
	private String comment;

	private static Map<Integer, UserStatusType> find = new HashMap<>();

	static {
		Arrays.stream(UserStatusType.values()).forEach(type -> find.put(type.statusCode, type));
	}

	UserStatusType(int statusCode, String comment) {
		this.statusCode = statusCode;
		this.comment = comment;
	}

	public int getStatusCode() {
		return statusCode;
	}

	public String getComment() {
		return comment;
	}

	public static UserStatusType get(int statusCode) {
		return find.get(statusCode);
	}
}

File: backend/src/main/kotlin/com/hst/scrapper/modules/user/domain/entity/UserStatusType.kt
package com.hst.scrapper.modules.user.domain.entity

import javax.persistence.AttributeConverter

/**
 * @author dlgusrb0808@gmail.com
 */
enum class UserStatusType(val statusCode: Int, val comment: String) {
    EMAIL_NOT_VERIFIED(1, ""),
    INACTIVE(2, ""),
    ACTIVE(3, "")
    ;

    companion object {
        val HOLDER = values().associateBy { it.statusCode }
        val default = EMAIL_NOT_VERIFIED
    }

    class Converter: AttributeConverter<UserStatusType, Int> {
        override fun convertToDatabaseColumn(attribute: UserStatusType?): Int {
             return attribute?.statusCode ?: default.statusCode
        }

        override fun convertToEntityAttribute(dbData: Int?): UserStatusType {
            return HOLDER.getOrDefault(dbData, default)
        }
    }
}